server/src/main/java/org/elasticsearch/index/store/Store.java

```diff
@@ -140,7 +139,11 @@
         @Override
         protected void closeInternal() {
             // close us once we are done
-            Store.this.closeInternal();
+            try {
+                Store.this.closeInternal();
+            } catch (IOException e) {
+                logger.warn("exception on closing store", e);
```

Comment:
Not sure what else we can do here. Currently some call-sites handle `IOException` by logging  others don't and would have some unexpected side-effects if the store was corrupted and failed to close. This seems like the best we can do.